### PR TITLE
Support & Types in Dotty (retry for rebuild!)

### PR DESCRIPTION
**Pull request by howyp**
Adds support for intersection types from Dotty, contributing to #269.

### CODE REVIEW

**Comment by jvican** [comment]
Amazing @howyp, thanks for the contribution :+1:

**Comment by jvican** [comment]
Could we label this as `hackathon` @xeno-by ?

File: scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala [hunk 1]
```diff
@@ -7,14 +7,14 @@ import scala.meta.dialects.Scala211
...
 class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
-  override def term(code: String)(implicit dialect: Dialect) = super.term(code).resetAllOrigins
-  override def pat(code: String)(implicit dialect: Dialect) = super.pat(code).resetAllOrigins
-  override def tpe(code: String)(implicit dialect: Dialect) = super.tpe(code).resetAllOrigins
-  override def topStat(code: String)(implicit dialect: Dialect) = super.topStat(code).resetAllOrigins
-  override def templStat(code: String)(implicit dialect: Dialect) = super.templStat(code).resetAllOrigins
-  override def blockStat(code: String)(implicit dialect: Dialect) = super.blockStat(code).resetAllOrigins
-  override def caseClause(code: String)(implicit dialect: Dialect) = super.caseClause(code).resetAllOrigins
-  override def source(code: String)(implicit dialect: Dialect) = super.source(code).resetAllOrigins
+  override def term(code: String)(implicit dialect: Dialect)       = super.term(code)(dialect).resetAllOrigins
```
Comment (jvican) [inline scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala]: Why is `dialect` explicit? IMO, being an implicit, it's redundant.

Comment (howyp) [inline scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala]: Yes, I was surprised by this - the `import scala.meta.dialects.Scala211` at the top of the file takes precedence over the implicit `dialect` parameter to the function when calling `super.tpe(...)`

Comment (jvican) [inline scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala]: Makes sense :+1:

**Comment by xeno-by** [comment]
@howyp Thanks for your excellent work! I'm back from vacation, and I'm glad to review your PR. 

Everything looks good apart from a minor thing. Could you update `notes/quasiquotes.md` like it's done in https://github.com/scalameta/scalameta/pull/475?

**Comment by xeno-by** [comment]
Superseded by https://github.com/scalameta/scalameta/pull/504

**Pull request closed without merging**